src/Halogen/ECharts.purs

```diff
@@ -118,8 +117,14 @@ eval (Init theme next) = do
     >>= traverse_ \el → do
       chart ← liftEff $ maybe EC.init EC.initWithTheme theme el
       H.modify _{ chart = Just chart }
+      H.subscribe
+        $ ES.eventSource (EE.listenAll chart)
+        ( Just <<< H.request <<< HandleEvent )
       H.raise Initialized
   pure next
+eval (HandleEvent evt reply) = do
+  H.raise $ EventRaised evt
+  pure $ reply H.Listening
```

Comment:
Subscriptions are always cleaned up by the Halogen runtime when a component is unmounted.